src/zarr/core/common.py

```diff
@@ -167,3 +169,24 @@ def parse_order(data: Any) -> Literal["C", "F"]:
     if data in ("C", "F"):
         return cast(Literal["C", "F"], data)
     raise ValueError(f"Expected one of ('C', 'F'), got {data} instead.")
+
+
+def _json_convert(o: Any) -> Any:
```

Comment:
Moved here so that I can use it in a couple of spots (but I think I forgot to replace the original usage of this).

All serialization stuff was a bit of a pain. I might look into cleaning that up a bit as a precursor.

Reply:
Hmm #1851 looks interesting. I started down a path of a pretty generic serialization / deserialization implementation at https://github.com/TomAugspurger/zarr-python/blob/feature/serde/src/zarr/_serialization.py. That's decoupled stores and mostly decoupled from Metadata objects. Though not necessarily using JSON might complicate things a bit.

This is a whole can of worms, which is why libraries like msgspec, cattrs, etc exist. I do think it's worth trying to make this work without taking a dependency on any of those first. I'll push that branch a little bit further and see how things work.